app/services/internal_proxy_proxy_selector.rb

```diff
@@ -0,0 +1,16 @@
+class InternalProxyProxySelector < ::RemoteExecutionProxySelector
```

Comment:
Get's us into nomenclature discussion, but I tend to like the `default` or `primary` or possibly even `master` terminology better. Since the smart proxy here is not internal to the system or the deployment (it is still running on a public interface).

Reply:
Katello uses `default`, guess I'll stick with it